lang_GENERIC_base/AST_generic.ml

```diff
@@ -1038,6 +1045,8 @@ and type_argument =
   (* Java only *)
   | TypeWildcard of tok (* '?' *) *
                     (bool wrap (* extends|super, true=super *) * type_) option
+  (* Rust *)
+  | TypeLifetime of ident
```

Comment:
It's a type parameter like the `'a` in `struct Ref<'a, T>`, not a concrete type, and is treated similar to types in generics. There is a good explanation [here](https://users.rust-lang.org/t/what-is-the-point-of-lifetime-parameters-in-struct-impl-blocks/14631/2). It encodes information about the references contained within the struct itself, separate from whether or not the struct is used in a reference type or not. So `struct Ref<'a, T>` means that "a concrete instantiation of `Ref` must go out of scope before any reference with lifetime `'a` goes out of scope." This remains true whether or not you see a type of `Ref` or `&Ref`. 

If you were specifying a function parameter, you could potentially do something like `fn my_function<'a, 'b>(ref: &'b Ref<'a, String>) { ... }` with multiple lifetimes. `'b` is the lifetime of the reference pointer of the `Ref` itself, and `'a` is the lifetime of the references inside the struct.

The main reason I added this is because you can't instantiate something with a type `'a`, because lifetimes aren't types.

Reply:
ok so if it's like a type parameter, we can use TyVar for that. We use it for polymorphic type in OCaml for example, in types like let (length: 'a list -> int) = fun ...